include/deal.II/cut/immersed_surface_quadrature.h

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright (C) 1998 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__cut_immersed_surface_quadrature
+#define dealii__cut_immersed_surface_quadrature
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/tensor.h>
+#include <deal.II/base/subscriptor.h>
+
+#include <vector>
+
+DEAL_II_NAMESPACE_OPEN
+namespace cut
+{
+
+  /**
+   * Defines a quadrature formula for integration over an oriented surface,
+   * $\hat{S}$, immersed in the unit cell. By immersed it is meant that the
+   * surface may intersect the unit cell in an arbitrary way. The quadrature
+   * formula is described by a set of quadrature points, $\hat{x}_q$, weights,
+   * $w_q$, and normalized surface normals, $\hat{n}_q$.
+   *
+   * We typically want to compute surface integrals in real space.
+   * A surface $S$ intersecting a cell $K$ in real space, can be mapped onto a
+   * surface $\hat{S}$ intersecting the unit cell $\hat{K}$.
+   * Thus a surface integral over $S$ in real space can be transformed to a
+   * surface integral over $\hat{S}$ according to
+   * @f[
+   * \int_{S} f(x) dS =
+   * \int_{S} f(x) |d\bar{S}| =
+   * \int_{\hat{S}} f(F_{K}(\hat{x})) \det(J) |\left( J^{-1} \right )^T d\hat{S}|.
+   * @f]
+   * This transformation is possible since the continuous surface elements
+   * are vectors: $d\bar{S}, d\hat{S} \in \mathbb{R}^{dim}$ which are parallel
+   * to the normals of $S$ and $\hat{S}$. So in order to compute the integral
+   * in real space one needs information about the normal to do the
+   * transformation.
+   *
+   * Thus, in addition to storing points and weights, this quadrature stores
+   * also the normalized normal for each quadrature point. This can be viewed
+   * as storing a discrete surface element,
+   * @f[
+   * \Delta \hat{S}_q := w_q \hat{n}_q \approx d\hat{S}(x_q),
+   * @f]
+   * for each quadrature point. The surface integral in real space would then be
+   * approximated as
+   * @f[
+   * \int_{S} f(x) dS \approx
+   * \sum_{q} f \left(F_{K}(x_{q}) \right) \det(J_q)
+   * |\left( J_q^{-1} \right)^T \hat{n}_q| w_q.
+   * @f]
+   *
+   * @image html immersed_surface_quadrature.svg
+   */
+  template<int dim>
+  class ImmersedSurfaceQuadrature : public Subscriptor
+  {
+  public:
+
+    /**
+     * Default constructor to initialize the quadrature with no quadrature
+     * points.
+     */
+    ImmersedSurfaceQuadrature();
```

Comment:
Since the implementation of this constructor is empty, you can just declare it as 
```
ImmersedSurfaceQuadrature() = default;
```
This way, you don't have to have an implementation in the .cc file at all.